model/aggregation.go

```diff
@@ -182,33 +183,44 @@ func (rc *realCluster) aggregateMetrics(target *InfoType, sources []*InfoType) e
 	}
 
 	// Create a map of []TimePoint as a timeseries accumulator per metric
-	newMetrics := make(map[string][]store.TimePoint)
+	newMetrics := make(map[string][]statstore.TimePoint)
 
 	// Reduce the sources slice with timeseries addition for each metric
 	for _, info := range sources {
-		for key, ts := range info.Metrics {
+		for key, ds := range info.Metrics {
 			_, ok := newMetrics[key]
 			if !ok {
 				// Metric does not exist on target map, create a new timeseries
-				newMetrics[key] = []store.TimePoint{}
+				newMetrics[key] = []statstore.TimePoint{}
 			}
 			// Perform timeseries addition between the accumulator and the current source
-			sourceTS := (*ts).Get(rc.timestamp, zeroTime)
-			newMetrics[key] = addMatchingTimeseries(newMetrics[key], sourceTS)
+			sourceDS := (*ds).Hour.Get(rc.timestamp, zeroTime)
+			newMetrics[key] = addMatchingTimeseries(newMetrics[key], sourceDS)
 		}
 	}
 
-	// Put all the new values in the TimeStores under target
+	// Put all the new values in the DayStores under target
 	for key, tpSlice := range newMetrics {
 		_, ok := target.Metrics[key]
 		if !ok {
 			// Metric does not exist on target InfoType, create TimeStore
-			newTS := rc.tsConstructor()
-			target.Metrics[key] = &newTS
+			// TODO(afein): configure epsilon
+			newDS := daystore.NewDayStore(100, rc.resolution)
```

Comment:
please do not review this PR yet, there are 650 failing assertions and a
set of refactoring changes that need to be put in first
On Aug 17, 2015 4:10 PM, "Vish Kannan" <notifications@github.com> wrote:

> In model/aggregation.go
> <https://github.com/kubernetes/heapster/pull/478#discussion_r37248065>:
>
> >  	for key, tpSlice := range newMetrics {
> >  		_, ok := target.Metrics[key]
> >  		if !ok {
> >  			// Metric does not exist on target InfoType, create TimeStore
> > -			newTS := rc.tsConstructor()
> > -			target.Metrics[key] = &newTS
> > +			// TODO(afein): configure epsilon
> > +			newDS := daystore.NewDayStore(100, rc.resolution)
>
> Do you have a follow up PR for epsilon? 100 defeats the whole effort.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/heapster/pull/478/files#r37248065>.
>